server/routes/newspaperArchive.ts

```diff
@@ -0,0 +1,61 @@
+import { Router } from 'express';
+import type { Request, Response } from 'express';
+import fetch from 'node-fetch';
+import { s3ConfigPromise } from '../awsIntegration';
+import { log } from '../log';
+import { withIdentity } from '../middleware/identityMiddleware';
+
+type NewspapersRequestBody = {
+	expires?: number; // defaults to 24 hours
+	'query-string'?: string;
+};
+
+// { url: "https://<subdomain>.newspapers.com/…?tpa=<token>" }
+type NewspapersResponseBody = {
+	url: string;
+};
+
+type NewspaperArchiveConfig = {
+	authString: string;
+};
+
+function base64(input: string) {
+	return Buffer.from(input).toString('base64');
+}
+
+const newspaperArchiveConfigPromise: Promise<
+	NewspaperArchiveConfig | undefined
+> = s3ConfigPromise<NewspaperArchiveConfig>('authString')('newspaper-archive');
+
+const router = Router();
+
+router.use(withIdentity(401));
+
+router.get('/auth', async (_req: Request, res: Response) => {
+	const config = await newspaperArchiveConfigPromise;
+	const authString = config?.authString;
+	if (authString === undefined) {
+		log.error(`Missing newspaper archive auth key`);
+		res.status(500).send();
```

Comment:
Possibly not the user experience we'd actually want

Reply:
see what you mean but the main thing for this MVP is that we get an alarm straight away if it happens